crates/router/src/routes/metrics/bg_metrics_collector.rs

```diff
@@ -0,0 +1,42 @@
+use storage_impl::redis::cache;
+
+const DEFAULT_BG_METRICS_COLLECTION_INTERVAL_IN_SECS: u64 = 15;
+
+pub fn spawn_metrics_collector(metrics_collection_interval_in_secs: &Option<u64>) {
+    let metrics_collection_interval = metrics_collection_interval_in_secs
+        .unwrap_or(DEFAULT_BG_METRICS_COLLECTION_INTERVAL_IN_SECS);
+
+    tokio::spawn(async move {
+        loop {
+            crate::gauge_metrics_for_imc!(
+                CONFIG_CACHE,
+                ACCOUNTS_CACHE,
+                ROUTING_CACHE,
+                CGRAPH_CACHE,
+                PM_FILTERS_CGRAPH_CACHE
+            );
```

Comment:
We'd have to include `DECISION_MANAGER_CACHE` and `SURCHARGE_CACHE` once #4814 is merged. How can we ensure that developers introducing new cache instances add metrics for them here: would including a comment in `crates/storage_impl/src/redis/cache.rs` be useful, or is there anything else that can be done?